pkg/api/validation/validation.go

```diff
@@ -687,11 +687,10 @@ func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
 
-	// TODO: allow change when bindings are properly decoupled from pods
```

Comment:
Otherwise kubelet would have to fill this, which doesn't make much sense since the host cannot change.

CC-ing @smarterclayton who wrote this function.